### ISSUE TITLE
once again (POS 10 in misc summons)

### ISSUE THREAD

**Issue by LordShade** [Masuzu/ZooeyBot#339]
got 5 days solid usage out of it. Today zoey ran into this and completely halted once again, on selecting a summon that was in the pos 10 slot of misc summons. Unsure if POS 9 affected like the previous bug, but probably likely. I am kinda annoyed that this keeps happening to me but it is not as frequent as before thanks to the mitigation effects (whatever they did?)

https://puu.sh/xNSwR/f41abed759.png

Tried to get the bot on track by clicking a bunch of various links to see if it would pick up again, but it appears to have halted in its track and will not proceed. Once again, the same issue rises where if i can scroll up fast enough to select the summon Zoey works fine. Am I really the only one that has this issue?

The most annoying bit is I went back to the Dark Tab to take a look if there really wasn't a good summon to use, and was confronted with 8 MLB Hades, and I have Hades listed as a summon to look for. Zoey just automatically went for the hardest option, a single Bahamut in misc slot 10 and screwed up. 

I do realize this bug is extremely, EXTREMELY hard to replicate, as it requires pure dumb luck to happen, heck it took 5 days to resurface...

**Comment by Masuzu** [comment]
Hi,
I wish I could replicate this bug and I do understand it can be really, really annoying indeed.
This is the mouse pointer settings I use on my laptop:
![](https://i.imgur.com/zzjN6YT.png)
My mouse scrolling speed is set to `MouseScrollSpeed=250` and summons at the very bottom of the list (whether in the misc or other tabs) are picked fine.
If I understand well, the click position after scrolling down is incorrect?
There is also a configuration parameter which configures the delay after scrolling down but it's not user overridable for now. It should be set on a value high enough for it to work fine even on less powerful computers but it seems it is not the case...

**Comment by LordShade** [comment]
POS 10 is in the middle of the list. Zoey scrolls all the way to the bottom and gets stuck. Each Summon tab holds I think 10, however Misc tab holds 20 summons due to the ability to place 2 summons in the misc slot.

https://puu.sh/xO8jh/bc4895e285.jpg here is an image I captured earlier (this time, my entire screen on my laptop) of where Zoey screws up, take note of the location of the scroll bar (all the way at the bottom, which makes me think that Zoey is thinking pos 20 in MISC = POS 10) I had to scroll a fair way up before I found the offending Bahamut summon and removed it from the summon list.

Seems like the easiest option to fix this annoyance is to just introduce the feature to ignore misc selection as a config option in summons category. However i don't know how much effort this will take considering I'm probably one of the only ones getting this. 

Also, MouseScrollSpeed=250 is what i've used since i had to replace all my config files on version 2.0.0 release. I have yet again reduced it to a lower number but I don't think it'll change anything.

**Comment by Masuzu** [comment]
Ok, thank you for the clarification. I will see what I can do later.

**Comment by sd5182487** [comment]
Same issue here. After couple of hours observing the bot action, I found myself having the same problem as PO, the bot thinks clicking the summon but actually it's out of the game frame.
shot here: https://imgur.com/a/zLrkM

**Comment by Masuzu** [comment]
Now that I think about it, if you use Viramate, it may reorder the summon list after Zooey retrieves the list of summons, resulting in this kind of issue.

**Comment by sahideler** [comment]
test

**Comment by LordShade** [comment]
@Masuzu I tried that in viramate, removed all the summon options in the extension that stops the re-ordering of summons, it still does it on occasion. Except lets add Position 11 to the list as well. 

Also upon counting the summon list down, I found that zoeybot is missing summons, When she is trying to click on pos 11, she is actually looking for the summon on pos 12 - (which causes her to keep scrolling trying to find it and she bugs the heck out), I've noticed this as a consistent issue for misc summons, so something is definitely wrong with selection here especially further down the list on misc. While summons near the top are of no issue to Zoey, there might be a desync issue somewhere as it scrolls down.
